apps/backoffice-v2/src/pages/Document/Document.page.tsx

```diff
@@ -0,0 +1,29 @@
+import { useParams, useSearchParams } from 'react-router-dom';
+import { useWorkflowQuery } from '../../domains/workflows/hooks/queries/useWorkflowQuery/useWorkflowQuery';
+import { Case } from '../Entity/components/Case/Case';
+
+export const DocumentLayout = () => {
+  const { entityId, documentId } = useParams();
+  const [searchParams] = useSearchParams();
+
+  const filterId = searchParams.get('filterId');
+  const { data: workflow, isLoading } = useWorkflowQuery({ workflowId: entityId, filterId });
+
+  if (isLoading) {
+    return;
+  }
+
+  const document = workflow.context.documents
+    .map(({ pages }) => pages)
+    .flat()
+    ?.find(({ ballerineFileId }) => ballerineFileId === documentId);
```

Comment:
Likely to break without optional chaining.
```suggestion
  const document = workflow?.context?.documents
    ?.map(({ pages }) => pages)
    ?.flat()
    ?.find(({ ballerineFileId }) => ballerineFileId === documentId);
```